CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+add_library(deflate::deflate ALIAS deflate)
+
+set_target_properties(deflate PROPERTIES VERSION 0.0.0 SOVERSION 0)
+
+target_sources(deflate PRIVATE ${LIBDEFLATE_SOURCES}
+                               ${LIBDEFLATE_HEADERS})
+
+# Compile options
+check_c_compiler_flag(-Wall DEFLATE_LINT_WALL)
+check_c_compiler_flag(-Wundef DEFLATE_LINT_WUNDEF)
+check_c_compiler_flag(-Wpedantic DEFLATE_LINT_PEDANTIC)
+check_c_compiler_flag(-Wdeclaration-after-statement DEFLATE_LINT_DECLARATION_AFTER_STATEMENT)
+check_c_compiler_flag(-Wmissing-prototypes DEFLATE_LINT_MISSING_PROTOTYPES)
+check_c_compiler_flag(-Wstrict-prototypes DEFLATE_LINT_STRICT_PROTOTYPES)
+check_c_compiler_flag(-Wvla DEFLATE_LINT_VLA)
+check_c_compiler_flag(-Wimplicit-fallthrough DEFLATE_LINT_IMPLICIT_FALLTHROUGH)
+
+set(lint-all $<$<BOOL:${DEFLATE_LINT_WALL}>:-Wall>)
+set(lint-undef $<$<BOOL:${DEFLATE_LINT_UNDEF}>:-Wundef>)
+set(lint-pedantic $<$<BOOL:${DEFLATE_LINT_PEDANTIC}>:-Wpedantic>)
+set(lint-declaration-after-statement $<$<BOOL:${DEFLATE_LINT_DECLARATION_AFTER_STATEMENT}>:-Wdeclaration-after-statement>)
+set(lint-missing-prototypes $<$<BOOL:${DEFLATE_LINT_MISSING_PROTOTYPES}>:-Wmissing-prototypes>)
+set(lint-strict-prototypes $<$<BOOL:${DEFLATE_LINT_STRICT_PROTOTYPES}>:-Wstrict-prototypes>)
+set(lint-vla $<$<BOOL:${DEFLATE_LINT_VLA}>:-Wvla>)
+set(lint-implicit-fallthrough $<$<BOOL:${DEFLATE_LINT_IMPLICIT_FALLTHROUGH}>:-Wimplicit-fallthrough>)
+
+target_compile_options(deflate PRIVATE ${lint-all})
+target_compile_options(deflate PRIVATE ${lint-undef})
+target_compile_options(deflate PRIVATE ${lint-pedantic})
+target_compile_options(deflate PRIVATE ${lint-declaration-after-statement})
+target_compile_options(deflate PRIVATE ${lint-missing-prototypes})
+target_compile_options(deflate PRIVATE ${lint-strict-prototypes})
+target_compile_options(deflate PRIVATE ${lint-vla})
+target_compile_options(deflate PRIVATE ${lint-implicit-fallthrough})
+
+# The CFLAGS environment variable is read and handled automatically by cmake
+
+if (BUILD_SHARED_LIBS)
+  target_compile_definitions(deflate PRIVATE _ANSI_SOURCE)
+  set_target_properties(deflate PROPERTIES C_VISIBILITY_PRESET hidden)
+
+  if (WIN32)
+    target_compile_definitions(deflate PUBLIC LIBDEFLATE_DLL)
+  endif()
+endif()
+
+if (FREESTANDING)
+  target_compile_definitions(deflate PRIVATE FREESTANDING)
+  target_compile_options(deflate PRIVATE -ffreestanding -nostdlib)
+endif()
+
+if (DEFLATE_TEST_SUPPORT__DO_NOT_USE)
+  target_compile_definitions(deflate PRIVATE TEST_SUPPORT__DO_NOT_USE)
+endif()
+
+target_include_directories(deflate PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<INSTALL_INTERFACE:include>
+)
+
+set_target_properties(deflate PROPERTIES PUBLIC_HEADER "libdeflate.h")
+
+install(TARGETS deflate
+        EXPORT deflate
+        ARCHIVE DESTINATION lib
+        PUBLIC_HEADER DESTINATION include/
+)
+
+# This is required to work with FetchContent
+install(EXPORT deflate
+        FILE deflate-config.cmake
```

Comment:
The existing pkg-config export is called libdeflate.pc, used via
- `dependency('libdeflate')` (meson)
- `pkg_check_modules(LIBDEFLATE libdeflate IMPORTED_TARGET GLOBAL)` (cmake)
- `pkg-config --cflags --libs libdeflate` (Makefile)


Hence, the cmake-config name must be libdeflate-config.cmake, used via:
- `find_package(libdeflate)` (cmake)
- `dependency('libdeflate')` (meson)


It is extremely important the names be consistent between pkg-config and cmake. e.g. Meson can look up this project as a dependency using multiple methods: pkg-config, cmake, checking for [a standalone ini file](https://mesonbuild.com/Wrap-dependency-system-manual.html#wrap-format) defining the location to download the dependency from...

However, in order for that to work, the names *must* be the same, or Meson will look for `libdeflate-config.cmake` when it is strangely named `deflate-config.cmake`.